File: app/src/main/java/mal/art/shopin/ViewModel/ShoppingListViewModel.java
package mal.art.shopin.ViewModel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import java.util.List;
import mal.art.shopin.Model.Product;
import mal.art.shopin.Repository.ProductRepository;

public class ShoppingListViewModel extends ViewModel {

  private ProductRepository repository;

  private String productName, productUnit;

  private int productQuantity;

  private MutableLiveData<List<Product>> shoppingList;
}

File: app/src/main/java/mal/art/shopin/viewModel/ShoppingListViewModel.kt
package mal.art.shopin.viewModel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.google.firebase.database.DataSnapshot
import mal.art.shopin.model.ProductModel
import mal.art.shopin.repository.ProductRepository

class ShoppingListViewModel(application: Application) : AndroidViewModel(application) {
  private val repository: ProductRepository = ProductRepository(application)
  private val shoppingListName : String? = null

  fun getShoppingList(shoppingListName: String) : LiveData<DataSnapshot?>? {
    return repository.getShoppingListLiveData(shoppingListName)
  }

  fun addProductToShoppingList(listName: String, productModel: ProductModel) {
    repository.addProductToParticularShoppingList(listName, productModel)
  }
}